src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -1332,16 +1332,51 @@ void MacroAssembler::movptr(Register r, uintptr_t imm64) {
 }
 
 // Macro to mov replicated immediate to vector register.
-//  Vd will get the following values for different arrangements in T
-//   imm32 == hex 000000gh  T8B:  Vd = ghghghghghghghgh
-//   imm32 == hex 000000gh  T16B: Vd = ghghghghghghghghghghghghghghghgh
-//   imm32 == hex 0000efgh  T4H:  Vd = efghefghefghefgh
-//   imm32 == hex 0000efgh  T8H:  Vd = efghefghefghefghefghefghefghefgh
-//   imm32 == hex abcdefgh  T2S:  Vd = abcdefghabcdefgh
-//   imm32 == hex abcdefgh  T4S:  Vd = abcdefghabcdefghabcdefghabcdefgh
-//   T1D/T2D: invalid
-void MacroAssembler::mov(FloatRegister Vd, SIMD_Arrangement T, uint32_t imm32) {
-  assert(T != T1D && T != T2D, "invalid arrangement");
+// imm64: only the lower 8/16/32 bits are considered for B/H/S type. That is,
+//        the upper 56/48/32 bits must be zeros for B/H/S type.
+// Vd will get the following values for different arrangements in T
+//   imm64 == hex 000000gh  T8B:  Vd = ghghghghghghghgh
+//   imm64 == hex 000000gh  T16B: Vd = ghghghghghghghghghghghghghghghgh
+//   imm64 == hex 0000efgh  T4H:  Vd = efghefghefghefgh
+//   imm64 == hex 0000efgh  T8H:  Vd = efghefghefghefghefghefghefghefgh
+//   imm64 == hex abcdefgh  T2S:  Vd = abcdefghabcdefgh
+//   imm64 == hex abcdefgh  T4S:  Vd = abcdefghabcdefghabcdefghabcdefgh
+//   imm64 == hex abcdefgh  T1D:  Vd = 00000000abcdefgh
+//   imm64 == hex abcdefgh  T2D:  Vd = 00000000abcdefgh00000000abcdefgh
+// Clobbers rscratch1
+void MacroAssembler::mov(FloatRegister Vd, SIMD_Arrangement T, uint64_t imm64) {
+  if (T == T1D || T == T2D) {
+    // To encode into movi, the 64-bit imm must be in the form of
+    // 'aaaaaaaabbbbbbbbccccccccddddddddeeeeeeeeffffffffgggggggghhhhhhhh'
+    // and encoded in "a:b:c:d:e:f:g:h".
+    bool can_encode = true;
+    uint64_t tmp = imm64;
+    uint64_t one_byte = 0;
+    for (int i = 0; i < 8; i++) {
+      one_byte = tmp & 0xFFULL;
+      if (one_byte != 0xFFULL && one_byte != 0) {
+        can_encode = false;
+        break;
+      }
+      tmp = tmp >> 8;
+    }
+
```

Comment:
This logic should be in a separate function.

Reply:
I think we need a `can_encode(imm, arrangement)` function.

Reply:
And then another function that actually does the arranging, and the generation of instructions calls those functions.

Reply:
Thanks for your review. I agree with that `can_encode(imm, arrangment)` function is better. My concern is that this JBS is just a bug fix for replicate2L_imm backend, and for other SIMD_Arrangment, I found that they can have some other choice for the code generation, but I didn’t touch them in this patch to keep it clear and small.  I list two examples below.

Example1:
```
        movi  v16.4s, #0x34
        orr v16.4s, #0x12, lsl #8

        vs

        mov w8, #0x1234
        dup v16.4s, w8
```

Example2:
```
        movi    v16.4s, #0x78
        orr     v16.4s, #0x56, lsl #8
        orr     v16.4s, #0x34, lsl #16
        orr     v16.4s, #0x12, lsl #24

        vs

        mov     w14, #0x5678
        movk    w14, #0x1234, lsl #16
        dup     v16.4s, w14
```

I'm considering to measure the performance and refine the mov macro assembler if it's necessary. `can_encode` can also be done in the refined work. What do you think?

Reply:
Sure. I'm looking at Neoverse V1 Optimization Guide, which suggests a fairly high cost for core - SIMD moves, and also only a 2 (of 4) of the SIMD pipelines can communicate with the integer registers. So I've got an idea.
Please feel free to do any reorganization later, if you like. It's just that the current organization makes it hard to follow, and thus hard to review.